packages/material-ui/src/InputLabel/InputLabel.d.ts

```diff
@@ -26,6 +26,4 @@ export type InputLabelClassKey =
   | 'filled'
   | 'outlined';
 
-declare const InputLabel: React.ComponentType<InputLabelProps>;
-
-export default InputLabel;
+export default class InputLabel extends React.Component<InputLabelProps> {}
```

Comment:
This is a general issue with our typings. Currently we technically don't allow any `ref` on the exported component but would allow `innerRef`. However `innerRef` does not work in many cases where we implemented the inner component as a function component. 

This change is only applied since we rely on it in https://github.com/mui-org/material-ui/blob/da515757f12f18cabcdb223ad513c1cc91a8342c/docs/src/pages/demos/text-fields/ComposedTextField.tsx#L70

The issue will be fully resolved once we switch to `@material-ui/styles` and proper ref forwarding.